File: starter_kits/Java/hlt/Dropoff.java
package hlt;

public class Dropoff extends Entity {
    public Dropoff(final PlayerId owner, final EntityId id, final Position position) {
        super(owner, id, position);
    }

    static Dropoff _generate(final PlayerId playerId) {
        final Input input = Input.readInput();

        final EntityId dropoffId = new EntityId(input.getInt());
        final int x = input.getInt();
        final int y = input.getInt();

        return new Dropoff(playerId, dropoffId, new Position(x, y));
    }
}

File: starter_kits/Kotlin/hlt/DropOff.kt
package hlt

class DropOff(owner: PlayerId, id: EntityId, position: Position) : Entity(owner, id, position) {
    companion object {

        internal fun generate(playerId: PlayerId): DropOff {
            val input = Input.readInput()

            val dropOffId = EntityId(input.nextInt)
            val x = input.nextInt
            val y = input.nextInt

            return DropOff(playerId, dropOffId, Position(x, y))
        }
    }
}
